### PR TITLE
CLOUDP-128685: Fix e2e tests

### PR DESCRIPTION
**Pull request by Sugar-pack**
### All Submissions:

Fix ginkgo label filter value.
Update operator-sdk version to 1.22.0. 
Remove observed generation check in case this check is meaningless.

* [x] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [ ] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if there is one).
* [ ] Update docs/release-notes/release-notes.md if your changes should be included in the release notes for the next release.

### CODE REVIEW

File: test/e2e/actions/steps.go [hunk 1]
```diff
@@ -71,6 +71,13 @@ func WaitProject(data *model.TestDataProvider, generation string) {
 	ExpectWithOffset(1, atlasProject.Status.ID).ShouldNot(BeNil(), "Kubernetes resource: Status has field with ProjectID")
 }
 
+func WaitProjectWithoutGenerationCheck(data *model.TestDataProvider) {
+	EventuallyWithOffset(1, kube.GetReadyProjectStatus(data), "15m", "10s").Should(Equal("True"), "Kubernetes resource: Project status `Ready` should be 'True'")
+	atlasProject, err := kube.GetProjectResource(data)
+	Expect(err).ShouldNot(HaveOccurred())
+	ExpectWithOffset(1, atlasProject.Status.ID).ShouldNot(BeNil(), "Kubernetes resource: Status has field with ProjectID")
```
Comment (fabritsius) [inline test/e2e/actions/steps.go:78]: The comment here is a little bit misleading as the field names do not match. Not critical thought:)

**Review by fabritsius — APPROVED** [review]
Looks Good 👍

File: test/e2e/actions/steps.go (hunk 1 above)
Comment (Sugar-pack) [reply test/e2e/actions/steps.go:78]: You're talking about line 78, did I understand you correctly?

Comment (Sugar-pack) [reply test/e2e/actions/steps.go]: Thanks. Done)

**Review by igor-karpukhin — APPROVED** [review]
Looks good 👍

**Pull request merged**